samples/net/openthread/ncp/src/main.c

```diff
@@ -0,0 +1,23 @@
+/* main.c - OpenThread NCP */
+
+/*
+ * Copyright (c) 2020 Tridonic GmbH & Co KG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(ot_br, LOG_LEVEL_DBG);
+
+#include <zephyr.h>
+
+#define APP_BANNER "***** OpenThread NCP on Zephyr %s *****"
+
+void main(void)
+{
+	LOG_INF(APP_BANNER, CONFIG_APPLICATION_VERSION);
+
+	while (1) {
+		k_sleep(MSEC_PER_SEC * 4U);
```

Comment:
Why not just exit main?

Reply:
See discussion in https://github.com/zephyrproject-rtos/zephyr/pull/23249#discussion_r387699909. I checked on hardware and it works. Also checked in code, it seems the main thread is marked non-essential after it has ended.

Fixed.